examples/hardware_encoding/main.go

```diff
@@ -88,10 +104,18 @@ func main() {
 	}
 	defer hardwareFramesContext.Free()
 
+	valid_sw_formats := hardwareFramesConstraints.ValidSoftwarePixelFormats()
+	if len(valid_sw_formats) == 0 {
+		log.Fatal("main: no valid software pixel formats")
+		return
+	}
+
 	// Get software pixel format
 	softwarePixelFormat := astiav.FindPixelFormatByName(*softwarePixelFormatName)
 	if softwarePixelFormat == astiav.PixelFormatNone {
 		log.Fatal("main: software pixel format not found")
+	} else if !slices.Contains(valid_sw_formats, softwarePixelFormat) {
+		log.Fatal("main: software pixel format not supported : ", softwarePixelFormat)
```

Comment:
Can you use `log.Fatalf("main: software pixel format not supported: %s", softwarePixelFormat)` instead?